src/gui/topbar_subwindows/military_subwindows/gui_build_unit_large_window.hpp

```diff
@@ -682,71 +244,106 @@ class build_unit_large_window : public window_element_base {
 			auto ptr = make_element_by_type<element_base>(state, id);
 			ptr->set_visible(state, false);
 			return ptr;
+		} else if(name.substr(0, 12) == "unit_folder_") {
+			auto ptr = make_element_by_type<unit_folder_button>(state, id);
+			auto value = std::stoi(std::string(name.substr(12)));
+			for(uint8_t i = 0; i < state.military_definitions.unit_base_definitions.size(); i++) {
+				if(state.military_definitions.unit_base_definitions[dcon::unit_type_id(i)].icon == value) {
+					ptr->unit_type = dcon::unit_type_id(i);
+					break;
+				}
+			}
+			switch(value) {
+			case 1:
```

Comment:
This won't work, because which values correspond to land or naval units could be changed by mods

Reply:
Admittedly, that was a problem with the previous version as well